src/main/java/link/myrecipes/api/service/MemberServiceImpl.java

```diff
@@ -60,7 +60,7 @@ public User readMember(int id) {
     public User createMember(UserRequest userRequest) {
 
         if (this.memberRepository.findByUsername(userRequest.getUsername()).isPresent()) {
-            throw new CustomValidationException("이미 사용중인 아이디입니다.", "username");
+            throw new DuplicateDataException(userRequest.getClass(), userRequest.getUsername());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code contains a potential information leakage in the error handling for the following call(s): &#39;UserRequest.getUsername()&#39; (line 63). You are handling this error data through: &#39;DuplicateDataException.&lt;init&gt;&#39; (line 63). Please use your organization-specific sensitive information handling loggers or redaction. [Learn more about information leakage](https://cwe.mitre.org/data/definitions/209.html) (external link).